datafusion-jit/src/jit.rs

```diff
@@ -51,6 +52,28 @@ impl Default for JIT {
             module,
         }
     }
+
+    #[cfg(target_arch = "aarch64")]
+    fn default() -> Self {
+        let mut flag_builder = settings::builder();
+        // On at least AArch64, "colocated" calls use shorter-range relocations,
+        // which might not reach all definitions; we can't handle that here, so
+        // we require long-range relocation types.
+        flag_builder.set("use_colocated_libcalls", "false").unwrap();
+        flag_builder.set("is_pic", "false").unwrap();
```

Comment:
perhaps worth linking to https://github.com/bytecodealliance/wasmtime/issues/2907 in the comment so we know to reenable pic when it's supported upstream.

Reply:
Good call